JitsiConference.js

```diff
@@ -2205,6 +2205,19 @@ JitsiConference.prototype.onIncomingCall = function(jingleSession, jingleOffer,
     }
 };
 
+/**
+ * Removes all tracks that don't match the criteria to be added to the peerconnection on init.
+ *
+ * @returns {void}
+ */
+JitsiConference.prototype._removeUnusedTracksOnInit = function() {
+    this.getLocalTracks().forEach(track => {
+        if (!this._shouldUseTrackOnInit(track)) {
+            this.removeTrack(track);
```

Comment:
Why are we trying to remove the track from the pc if it wasn't added in the first place ? This effectively results in replaceTrack(null, null) being called since the old track wasn't added to the conference. 

Can we call `onLocalTrackRemoved` or fire `JitsiConferenceEvents.TRACK_REMOVED` directly ?

Reply:
You say we don't need any of those:
```
if (oldTrackBelongsToConference && !oldTrack.isMuted() && !newTrack) {
                oldTrack._sendMuteStatus(true);
            }
            oldTrackBelongsToConference && this.onLocalTrackRemoved(oldTrack);
            newTrack && this._setupNewTrack(newTrack);

            // Send 'VideoTypeMessage' on the bridge channel when a video track is added/removed.
            if ((oldTrackBelongsToConference && oldTrack?.isVideoTrack()) || newTrack?.isVideoTrack()) {
                this._sendBridgeVideoTypeMessage(newTrack);
            }
            this._updateRoomPresence(this.getActiveMediaSession());
            if (newTrack !== null && (this.isMutedByFocus || this.isVideoMutedByFocus)) {
                this._fireMuteChangeEvent(newTrack);
            }
```

Reply:
Well, none of those get executed since it fails `oldTrackBelongsToConference` check. The tracks were never added to the conference. So if your intention is to remove the tracks from RTC and fire the TRACK_REMOVED event, you should call `onLocalTrackRemoved` directly.

Reply:
Scratch that ... the old track is added to the conference but not to the pc.